dspace-server-webapp/src/test/java/org/dspace/app/rest/AuthorizationRestRepositoryIT.java

```diff
@@ -0,0 +1,1170 @@
+                    .andExpect(status().isNotFound());
+
+    }
+
+    @Test
+    /**
+     * Verify that an exception in the feature check will be reported back
+     *
+     * @throws Exception
+     */
+    public void findOneInternalServerErrorTest() throws Exception {
+        Site site = siteService.findSite(context);
+        SiteRest siteRest = converterService.toRest(site, DefaultProjection.DEFAULT);
+        // define two authorizations that we know will throw exceptions
+        Authorization authAdminSite = new Authorization(admin, alwaysException, siteRest);
+        Authorization authNormalUserSite = new Authorization(eperson, alwaysException, siteRest);
+
+        String adminToken = getAuthToken(admin.getEmail(), password);
+        String epersonToken = getAuthToken(eperson.getEmail(), password);
+
+        getClient(adminToken).perform(get("/api/authz/authorizations/" + authAdminSite.getID()))
+                    .andExpect(status().isInternalServerError());
+
+        getClient(epersonToken).perform(get("/api/authz/authorizations/" + authNormalUserSite.getID()))
+                    .andExpect(status().isInternalServerError());
+    }
+
+    @Test
+    /**
+     * Verify that the search by object works properly in allowed scenarios:
+     * - for an administrator
+     * - for an administrator that want to inspect permission of the anonymous users or another user
+     * - for a logged-in "normal" user
+     * - for anonymous
+     * 
+     * @throws Exception
+     */
+    public void findByObjectTest() throws Exception {
+        Site site = siteService.findSite(context);
+        SiteRest siteRest = converterService.toRest(site, converterService.getProjection(DefaultProjection.NAME));
+        String siteUri = utils.linkToSingleResource(siteRest, "self").getHref();
+
+        // disarm the alwaysThrowExceptionFeature
+        configurationService.setProperty("org.dspace.app.rest.authorization.AlwaysThrowExceptionFeature.turnoff", true);
+        // verify that it works for administrators
+        String adminToken = getAuthToken(admin.getEmail(), password);
+        getClient(adminToken).perform(get("/api/authz/authorizations/search/object")
+                .param("uri", siteUri)
+                .param("eperson", admin.getID().toString()))
+            .andExpect(status().isOk())
+            // there are at least 3: alwaysTrue, trueForAdministrators and trueForLoggedUsers
+            .andExpect(jsonPath("$._embedded.authorizations", Matchers.hasSize(greaterThanOrEqualTo(3))))
+            .andExpect(jsonPath("$._embedded.authorizations", Matchers.everyItem(
+                    Matchers.anyOf(
+                            JsonPathMatchers.hasJsonPath("$.type", is("authorization")),
+                            JsonPathMatchers.hasJsonPath("$._embedded.feature",
+                                    Matchers.not(Matchers.anyOf(
```

Comment:
This tests which features are not embedded, but doesn't verify which features are embedded. The same applies for tests below
Also, I don't expect anything to be embedded since no projection is requested

Reply:
the reason why projection was not requested is because it was introduced after than these test were written. I have updated them now.
I cannot test that the expected features are included because we don't know all of them in advance. We will add more and more features soon and I prefer to avoid the need to update such test for each new feature. Instead, it would be sufficient to verify that what is returned match the request.

Reply:
I noticed the `$._embedded.feature` is currently still not verifying which data is present (only which data is not present). The test would be more reliable if it can verify the expected features are present

Reply:
fixed

Reply:
@abollini where has this been fixed?

Reply:
sorry, you are right the check were still missing. Added here https://github.com/DSpace/DSpace/pull/2663/commits/59b7716725588fafc44c988a8517aab022b6fc4e

Reply:
Thanks for the updates, that's really helpful.
I think you missed one though: https://github.com/DSpace/DSpace/blob/59b7716725588fafc44c988a8517aab022b6fc4e/dspace-server-webapp/src/test/java/org/dspace/app/rest/AuthorizationRestRepositoryIT.java#L536